spec/controllers/idv/finance_controller_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+describe Idv::FinanceController do
+  describe 'before_actions' do
+    it 'includes authentication before_action' do
+      expect(subject).to have_actions(
+        :before,
+        :confirm_two_factor_authenticated,
+        :confirm_idv_session_started
+      )
+    end
+  end
+
+  describe '#create' do
+    before do
+      allow(subject).to receive(:confirm_two_factor_authenticated).and_return(true)
+      allow(subject).to receive(:confirm_idv_session_started).and_return(true)
+      allow(subject).to receive(:idv_session).and_return(params: {})
+    end
+
+    context 'when form is invalid' do
+      render_views
```

Comment:
Rendering views in controller specs is expensive. I don't think we need to test every error message in a controller spec. I think it's enough to have a single test for the failure scenario that tests that the `new` action is rendered:
```ruby
expect(response).to render_template :new
```
I think feature tests are better suited for error messages, especially since we want to test the client-side error messages since most people will have JS enabled.